test/cpp/api/functional.cpp

```diff
@@ -541,6 +541,40 @@ TEST_F(FunctionalTest, PReLU) {
   ASSERT_TRUE(torch::allclose(y, y_exp));
 }
 
+TEST_F(FunctionalTest, Normalize) {
+  const auto expected = torch::tensor(
+    {{{0.00000000, 0.10000000, 0.2000, 0.30000000, 0.40000000},
+      {0.14285715, 0.17142858, 0.2000, 0.22857143, 0.25714287}}}, torch::requires_grad().dtype(torch::kFloat));
+  { // Test #1 
+    auto input = torch::tensor({{{0, 1, 2, 3, 4}, {5, 6, 7, 8, 9}}}, torch::dtype(torch::kFloat).requires_grad(true));
+    auto norm = F::normalize(input, NormalizeOptions().p(1).dim(-1));
+    
+    // reduce to scalar to call .backward()
+    torch::Tensor s = norm.sum();
+    s.backward();
+
+    ASSERT_EQ(s.ndimension(), 0);
+    ASSERT_EQ(input.grad().numel(), 10);
+    ASSERT_TRUE(torch::allclose(norm, expected));
+  }
+
+  { // Test #2 Check with non-null output parameter
+    auto input = torch::tensor({{{0, 1, 2, 3, 4}, {5, 6, 7, 8, 9}}}, torch::dtype(torch::kFloat));
+    auto output = torch::randn({1,2,5}, torch::dtype(torch::kFloat));
+    F::normalize(input, NormalizeOptions().p(1).dim(-1), output);
+    
+    ASSERT_TRUE(torch::allclose(output, expected));
+  }
+  
+  { // Test #3 Base case of scalar tensor
+    auto input = torch::randn({}, torch::requires_grad());
+    torch::Tensor norm = F::normalize(input, NormalizeOptions().p(2));
```

Comment:
I suspect that we should change the options args here:
```suggestion
    torch::Tensor norm = F::normalize(input, NormalizeOptions().p(1).dim(-1));
```
which matches the Python version and should fixes the test errors :D

Reply:
Thanks for catching this, sorry for the messy workflow I'm building the source on a caveman laptop right now or I would have checked the tests locally before pushing them (working on changing that though)